api/src/main/java/com/xwiki/googleapps/GoogleAppsException.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.googleapps;
+
+import java.io.IOException;
+
+import org.xwiki.stability.Unstable;
+
+import com.xpn.xwiki.XWikiException;
+
+/**
+ * Generic class to denote an exception condition within the GoogleApps code. Wraps XWikiException and IOException.
+ *
+ * @version $Id$
+ * @since 3.0
+ */
+public class GoogleAppsException extends RuntimeException
```

Comment:
Why ``RuntimeException`` and not ``Exception``? Since you have methods throwing it then it means its an "expected" exception, so code calling such a method should handle it.